src/Gang/Gang.ts

```diff
@@ -301,9 +308,22 @@ export class Gang {
     }
   }
 
-  canRecruitMember(): boolean {
-    if (this.members.length >= GangConstants.MaximumGangMembers) return false;
-    return this.respect >= this.respectForNextRecruit();
+  canRecruitMember(): { isSuccess: boolean; errorType?: RecruitmentErrorType } {
```

Comment:
Rather than have a compound type here, I think it would make more sense to return `RecruitmentResult`, which would be your current error enum plus "Success".